### PR TITLE
Identify raid enclosure

### PR DESCRIPTION
**Pull request by cholcombe973**

### CODE REVIEW

**Comment by cholcombe973** [comment]
@sdandam when you get a sec could you look at this?  I'm not entirely happy with the Option<String>'s on the `Enclosure` struct but I also don't know what the possible values are to use for an enum.

File: src/lib.rs [hunk 1]
```diff
@@ -918,18 +934,13 @@ fn get_serial(device: &libudev::Device) -> Option<String> {
...
-            match value.as_ref() {
-                "btrfs" => FilesystemType::Btrfs,
-                "xfs" => FilesystemType::Xfs,
-                "ext4" => FilesystemType::Ext4,
-                _ => FilesystemType::Unknown,
-            }
+            FilesystemType::from_str(&value)
         }
-        None => FilesystemType::Unknown,
+        None => Ok(FilesystemType::Unknown),
```
Comment (sdandam) [inline src/lib.rs:943]: I am wondering if this should be an error instead of Unknown?

File: src/lib.rs [hunk 2]
```diff
@@ -1312,6 +1440,24 @@ pub fn get_raid_info() -> BlockResult<Vec<ScsiInfo>> {
                             s.scsi_type = ScsiDeviceType::from_str(
                                 fs::read_to_string(&scsi_entry.path())?.trim(),
                             )?;
+                        } else if scsi_entry.file_name() == OsStr::new("block") {
+                            let block_path = path.join("block");
+                            if block_path.exists() {
+                                let block_entry = read_dir(&block_path)?.next();
+                                // There should only be 1 entry in this directory
```
Comment (sdandam) [inline src/lib.rs]: Is it an error if there are more than one entries in this directory?

**Review by sdandam — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (cholcombe973) [reply src/lib.rs:943]: Well udev is weird and fields aren't always shown.  It's just like the `/sys/bus/devices` filesystem.  It's hard to tell when entries might be there.

File: src/lib.rs (hunk 2 above)
Comment (cholcombe973) [reply src/lib.rs]: I think so yeah.  That would mean the disk is mapped to 2 block device names like `/dev/sda` and `/dev/sdb`

**Pull request merged**